hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/Router.java

```diff
@@ -291,22 +296,30 @@ public static String getProxyHostAndPort(Configuration conf) {
   public static void main(String[] argv) {
     Configuration conf = new YarnConfiguration();
     Thread
-        .setDefaultUncaughtExceptionHandler(new YarnUncaughtExceptionHandler());
+            .setDefaultUncaughtExceptionHandler(new YarnUncaughtExceptionHandler());
     StringUtils.startupShutdownMessage(Router.class, argv, LOG);
     Router router = new Router();
     try {
-
-      // Remove the old hook if we are rebooting.
-      if (null != routerShutdownHook) {
-        ShutdownHookManager.get().removeShutdownHook(routerShutdownHook);
+      GenericOptionsParser hParser = new GenericOptionsParser(conf, argv);
+      argv = hParser.getRemainingArgs();
+      if (argv.length > 1) {
```

Comment:
Don't we have better parsing?

Reply:
Thanks for your suggestion! I will improve this part of the code.